pkg/aws/services/vpclattice.go

```diff
@@ -181,51 +183,81 @@ func (d *defaultLattice) ListServiceNetworkServiceAssociationsAsList(ctx context
 func (d *defaultLattice) FindServiceNetwork(ctx context.Context, name string, optionalAccountId string) (*ServiceNetworkInfo, error) {
 	input := vpclattice.ListServiceNetworksInput{}
 
-	for {
-
-		resp, err := d.ListServiceNetworksWithContext(ctx, &input)
-		if err != nil {
-			return nil, err
-		}
-
-		for _, r := range resp.Items {
+	var innerErr error
+	var snMatch *ServiceNetworkInfo
+	err := d.ListServiceNetworksPagesWithContext(ctx, &input, func(page *vpclattice.ListServiceNetworksOutput, lastPage bool) bool {
+		for _, r := range page.Items {
 			if aws.StringValue(r.Name) != name {
 				continue
 			}
 			acctIdMatches, err1 := accountIdMatches(optionalAccountId, *r.Arn)
 			if err1 != nil {
-				return nil, err1
+				innerErr = err1
+				return false
 			}
 			if !acctIdMatches {
-				glog.V(6).Infoln("ServiceNetwork found but does not match account id ", name, r.Arn, optionalAccountId)
+				glog.V(6).Infoln("ServiceNetwork found but does not match account id", name, r.Arn, optionalAccountId)
 				continue
 			}
 
-			glog.V(6).Infoln("Found ServiceNetwork ", name, r.Arn, optionalAccountId)
-
+			glog.V(6).Infoln("Found ServiceNetwork", name, r.Arn, optionalAccountId)
 			tagsInput := vpclattice.ListTagsForResourceInput{
 				ResourceArn: r.Arn,
 			}
 
 			tagsOutput, err2 := d.ListTagsForResourceWithContext(ctx, &tagsInput)
 			if err2 != nil {
-				return nil, err2
+				innerErr = err2
+				return false
 			}
 
-			return &ServiceNetworkInfo{
+			snMatch = &ServiceNetworkInfo{
 				SvcNetwork: *r,
 				Tags:       tagsOutput.Tags,
-			}, nil
+			}
+			return false
 		}
 
-		if resp.NextToken == nil {
-			break
+		return true
+	})
+
+	if innerErr != nil {
+		return nil, innerErr
+	}
+	if err != nil {
+		return nil, err
+	}
+	if snMatch == nil {
+		glog.V(6).Infoln("Service network for account not found ", name, optionalAccountId)
+		return nil, NewNotFoundError("Service network", name)
+	}
+
+	return snMatch, nil
+}
+func (d *defaultLattice) FindService(ctx context.Context, nameProvider LatticeServiceNameProvider) (*vpclattice.ServiceSummary, error) {
+	serviceName := nameProvider.LatticeServiceName()
+	input := vpclattice.ListServicesInput{}
+
+	var svcMatch *vpclattice.ServiceSummary
+	err := d.ListServicesPagesWithContext(ctx, &input, func(page *vpclattice.ListServicesOutput, lastPage bool) bool {
+		for _, svc := range page.Items {
```

Comment:
Does FindService() also need to check:

```
acctIdMatches, err1 := accountIdMatches(optionalAccountId, *r.Arn)

if !acctIdMatches {
    continue
}
``` 

in case in the future we support RAM share lattice Service in the controller and one current account Service and one foreign Service have the same name. (check accountId at least not harmful?)